### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by alvaroiporre**
## Project requirements
In this PR I made the following:

- I added Redux Toolkit using (npm install react-redux @reduxjs/toolkit).
- I created a directory that will contain all your Redux logic (/src/redux).
- I configured a Redux store (/src/redux/store.js).
- I defined a slice of state for books that:
  - Includes an array of books (initial state: empty array)
  - Includes a reducer that adds a book
  - Includes a reducer that removes a book
- I defined a slice of state for categories that:
  - Includes an array of categories (initial state: empty array)
  - Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)
- I structured my application files using a "feature folder" approach and used the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for my Redux files. My files/folder structure look like this:
``` sh
/src
    |---/components
    |---/redux
        |--- /books
                | booksSlice.js
        |--- /categories
                | categoriesSlice.js
        | store.js
    | index.js
```

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+export const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkCategory: (state) => {
+      state.categories = 'Under construction';
+    },
+  },
+});
```
Comment (levy002) [inline src/redux/categories/categoriesSlice.js]: - [ ] Kindly make sure in the `checkCategory`, you are changing the right property in the state object. You changed the `state.categories` while in your `initialState` object you only have the `value` property. Kindly fix this.

```suggestion
const initialState = {
  categories: [],
};

export const categoriesSlice = createSlice({
  name: 'categories',
  initialState,
  reducers: {
    checkCategory: (state) => {
      state.categories = 'Under construction';
    },
  },
});
```

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+};
+
+export const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      const newBook = action.payload;
+      state.value.push(newBook);
+    },
```
Comment (levy002) [inline src/redux/books/booksSlice.js]: - [OPTIONAL] I highly recommend avoiding using the `push` method which mutates the array while adding a new book, instead you can use the Spread operator.

```suggestion
    addBook: (state, action) => {
      const newBook = action.payload;
      state.value = [...state.value, newBook];
    },
```

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @alvaroiporre  , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- NO linter error :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [OPTIONAL] Kindly make sure to edit or remove the broken link of your `Twitter` account because when we click on it we go a message saying that the `account doesn't exist`. Kindly fix it to keep your Readme clean and professional.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alvaroiporre** [comment]
Thank you so much @levy002, I appreciate your advice to improve my code.

**Review by abdallahmalima — APPROVED** [review]
Hi @alvaroiporre , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Setup `react-redux`✔️
- Use ducks pattern✔️
- Imports `configureStore`✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**